### PR TITLE
 Add double quotes to return tag

### PR DESCRIPTION
**Pull request by DamianSzymanski**
Fix issue with git checkout when tag contains special characters

### CODE REVIEW

**Comment by BanzaiMan** [comment]
What are problematic tags that this PR is supposed to solve?

**Comment by DamianSzymanski** [comment]
@BanzaiMan https://github.com/travis-ci/bugs-questions-comments-ideas/issues/129

**Comment by BanzaiMan** [comment]
Ah, I see. Could you add a spec for that problematic value, then? I tend to think, though, that we want to shell escape the tags properly rather than quoting with `%Q{""}`. We need to test it, to be sure.

File: lib/travis/build/git/clone.rb [hunk 1]
```diff
@@ -101,8 +101,8 @@ def checkout
           end
 
           def checkout_ref
-            return 'FETCH_HEAD' if data.pull_request
-            return data.tag     if data.tag
+            return 'FETCH_HEAD'      if data.pull_request
+            return %Q{"#{data.tag}"} if data.tag
```
Comment (cotsog) [inline lib/travis/build/git/clone.rb]: Could we call the `tag` method instead i.e. https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/git/clone.rb#L140-L142 ?

**Comment by DamianSzymanski** [comment]
@BanzaiMan I updated to shellescape, please check
@cotsog i am not sure about def tag we will have double if
Here is the build:
https://staging.travis-ci.org/DamianSzymanski/ddd/builds/739212#L180

File: lib/travis/build/git/clone.rb [hunk 2]
```diff
@@ -101,8 +101,8 @@ def checkout
           end
 
           def checkout_ref
-            return 'FETCH_HEAD' if data.pull_request
-            return data.tag     if data.tag
+            return 'FETCH_HEAD'         if data.pull_request
+            return data.tag.shellescape if data.tag
```
Comment (BanzaiMan) [inline lib/travis/build/git/clone.rb]: @DamianSzymanski I think what @cotsog is saying is that this line can be rewritten to:

```ruby
return tag if data.tag
```

**Comment by DamianSzymanski** [comment]
@BanzaiMan makes sense, please check
@cotsog Thanks

**Pull request merged**